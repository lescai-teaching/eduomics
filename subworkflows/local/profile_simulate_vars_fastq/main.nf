include { PYCONVERTOSIM      } from '../../../modules/local/pyconvertosim/main'
include { SIMUSCOP_SIMUREADS } from '../../../modules/local/simuscop/simureads/main'

workflow PROFILE_SIMULATE_VARS_FASTQ {

    take:
    vcf_benign // channel: [mandatory] [[meta], vcf_benign]
    vcf_patho  // channel: [mandatory] [[meta], vcf_patho]
    simprofile // channel: [mandatory] [[meta], profile]
    fasta_fai  // channel: [mandatory] [fasta, fai]
    capture    // channel: [mandatory] [capture_500pad]

    main:

    ch_versions = Channel.empty()

    PYCONVERTOSIM( vcf_benign, vcf_patho )
    ch_versions = ch_versions.mix(PYCONVERTOSIM.out.versions.ifEmpty([]))

    variants_to_inject = PYCONVERTOSIM.out.combined_variations.flatMap { m, files ->
                                                files.collect { file ->
                                                def var = file.getName().split('_')[2]
                                                def newmeta = m + [simulatedvar: "${var}"]
                                                return [newmeta, file]
                                                } }
    ch_variants_to_inject = params.istest
        ? variants_to_inject.take(params.test_limit)
        : variants_to_inject

    variants_to_inject.dump(tag: 'variants to inject')
    ch_variants_to_inject.dump(tag: 'variants selected for injection')

    SIMUSCOP_SIMUREADS(
        simprofile,
        fasta_fai,
        ch_variants_to_inject,
        capture
    )
    ch_versions = ch_versions.mix(SIMUSCOP_SIMUREADS.out.versions.ifEmpty([]))


    emit:
    simreads = SIMUSCOP_SIMUREADS.out.reads    // channel: [ val(meta), [ "control_1.fq.gz", "control_2.fq.gz", "case_1.fq.gz", "case_2.fq.gz" ] ]
    versions = ch_versions                     // channel: [ versions.yml ]
}

