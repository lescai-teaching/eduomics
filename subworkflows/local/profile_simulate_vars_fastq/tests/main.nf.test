// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core subworkflows test profile_simulate_vars_fastq
nextflow_workflow {

    name "Test Subworkflow PROFILE_SIMULATE_VARS_FASTQ"
    script "../main.nf"
    workflow "PROFILE_SIMULATE_VARS_FASTQ"

    tag "subworkflows"
    tag "subworkflows_"
    tag "subworkflows/profile_simulate_vars_fastq"
    // TODO nf-core: Add tags for all modules used within this subworkflow. Example:
    tag "samtools"
    tag "samtools/sort"
    tag "samtools/index"


    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("sarscov2 - bam - single_end") {

        when {
            workflow {
                """
                // TODO nf-core: define inputs of the workflow here. Example:
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                ]
                input[1] = [
                    [ id:'genome' ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }
    }
}
