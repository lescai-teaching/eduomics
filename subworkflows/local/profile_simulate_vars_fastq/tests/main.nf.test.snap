{
    "Simulate reads - profile - two pathological variants": {
        "content": null,
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-31T10:00:57.002992232"
    }
}